apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/assessmentQuestion.ts

```diff
@@ -1,78 +1,101 @@
-const express = require('express');
-const router = express.Router();
-const asyncHandler = require('express-async-handler');
-const error = require('@prairielearn/error');
-const sqldb = require('@prairielearn/postgres');
+import * as express from 'express';
+import asyncHandler = require('express-async-handler');
+import * as error from '@prairielearn/error';
+import { loadSqlEquiv, queryAsync, queryRows } from '@prairielearn/postgres';
+
+import * as manualGrading from '../../../lib/manualGrading';
+import { InstanceQuestionSchema } from '../../../lib/db-types';
+import { z } from 'zod';
 
-const manualGrading = require('../../../lib/manualGrading');
+const router = express.Router();
+const sql = loadSqlEquiv(__filename);
 
-const sql = sqldb.loadSqlEquiv(__filename);
+const InstanceQuestionRowSchema = InstanceQuestionSchema.extend({
+  modified_at: z.string(),
+  assessment_open: z.boolean(),
+  uid: z.string().nullable(),
+  assigned_grader_name: z.string().nullable(),
+  last_grader_name: z.string().nullable(),
+  max_points: z.number().nullable(),
+  max_auto_points: z.number().nullable(),
+  max_manual_points: z.number().nullable(),
+  user_or_group_name: z.string().nullable(),
+  open_issue_count: z.number().nullable(),
+});
 
 router.get(
   '/',
-  asyncHandler(async (req, res, next) => {
+  asyncHandler(async (req, res) => {
     if (!res.locals.authz_data.has_course_instance_permission_view) {
-      return next(error.make(403, 'Access denied (must be a student data viewer)'));
+      throw error.make(403, 'Access denied (must be a student data viewer)');
     }
-    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+    res.render(__filename.replace(/\.ts$/, '.ejs'), res.locals);
```

Comment:
This won't work when this is transpiled into JS, you need this instead:

```suggestion
    res.render(__filename.replace(/\.(js|ts)$/, '.ejs'), res.locals);
```

Existing page where we do this:

https://github.com/PrairieLearn/PrairieLearn/blob/13a7cb2b4495baa894193998f459d024fd2244e3/apps/prairielearn/src/pages/publicQuestionPreview/publicQuestionPreview.ts#L83